Universe files generation for multiple markets

jhonabreul opened the pull request:
Allow universe files generation for multiple markets at a time

jhonabreul committed:
Allow universe files generation for multiple markets at a time

jhonabreul committed:
Trigger GH actions on pull requests

jhonabreul committed:
Cleanup

Martin-Molinero commented on Lean.DataSource.DerivativeUniverseGenerator/Program.cs:
```diff
@@ -46,47 +50,69 @@ public abstract class Program
 
         protected abstract DerivativeUniverseGenerator GetUniverseGenerator(SecurityType securityType, string market, string dataFolderRoot,
-            string outputFolderRoot, DateTime processingDate);
+            string outputFolderRoot, DateTime processingDate, IDataProvider dataProvider, ZipDataCacheProvider dataCacheProvider,
```
Minor but `ZipDataCacheProvider` could just better be `IDataCacheProvider`

jhonabreul committed:
Refactor GetUniverseGenerator method signatures

Updated GetUniverseGenerator method signatures in Program.cs to use IDataCacheProvider instead of ZipDataCacheProvider
